source/eventHandler.py

```diff
@@ -253,24 +250,39 @@ def isMenuItemOfCurrentFocus(self) -> bool:
 		The focus event order after activating the menu item's sub menu is (submenu item, submenu).
 		"""
 		from NVDAObjects import IAccessible
+		from comInterfaces import IAccessible2Lib as IA2
+
 		lastFocus = api.getFocusObject()
-		_isMenuItemOfCurrentFocus = (
-			self._obj.parent
-			and isinstance(self._obj, IAccessible.IAccessible)
+		
+		# This case can only occur when:
+		# 1. the old and new focus targets are instances of IAccessible; and
+		# 2. the old focus is a menuitem, menuitemradio or menuitemcheckbox; and
+		# 3. the new focus is a menu; and
+		# 4. the old focus has a parent.
+		if not (
+			isinstance(self._obj, IAccessible.IAccessible)
 			and isinstance(lastFocus, IAccessible.IAccessible)
-			and self._obj.IAccessibleRole == oleacc.ROLE_SYSTEM_MENUITEM
+			and self._obj.IAccessibleRole in (
+				oleacc.ROLE_SYSTEM_MENUITEM, IA2.IA2_ROLE_CHECK_MENU_ITEM, IA2.IA2_ROLE_RADIO_MENU_ITEM)
 			and lastFocus.IAccessibleRole == oleacc.ROLE_SYSTEM_MENUPOPUP
-			and self._obj.parent == lastFocus
-		)
-		if _isMenuItemOfCurrentFocus:
-			# Change this to log.error for easy debugging
-			log.debugWarning(
-				"This parent menu was not announced properly, "
-				"and should have been focused before the submenu item.\n"
-				f"Object info: {self._obj.devInfo}\n"
-				f"Object parent info: {self._obj.parent.devInfo}\n"
-			)
-		return _isMenuItemOfCurrentFocus
+			and self._obj.parent):
+			return False
+
+		# Check that the old focus is a descendant of the new focus.
+		ancestor = self._obj.parent
+		while ancestor is not None:
+			if ancestor == lastFocus:
+				log.debugWarning(

```

Comment:
can you please keep this as an f-string? We generally prefer these to the older style of string formats

Reply:
I agree that F-strings are prefferable. However, f-strings are eagerly evaluated, meaning this will be evaluated even if it isn't being output to log, which is potentially bad for performance. If that's a trade-off that we're willing to make, I'm happy to change it, but this version is theoretically more performant.

Reply:
yes, I don't think this is a worthwhile performance concern and we prefer to use modern syntax where possible